jupyter/src/incremental_learning/storage.py

```diff
@@ -0,0 +1,45 @@
+# Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+# or more contributor license agreements. Licensed under the Elastic License
+# 2.0 and the following additional limitation. Functionality enabled by the
+# files subject to the Elastic License 2.0 may only be used in production when
+# invoked by an Elasticsearch process with a license key installed that permits
+# use of machine learning features. You may not use this file except in
+# compliance with the Elastic License 2.0 and the foregoing additional
+# limitation.
+
+from google.cloud import storage
+from .config import bucket_name, datasets_dir, configs_dir, logger
+
+def download_dataset(dataset_name):
+    storage_client = storage.Client()
+    bucket = storage_client.bucket(bucket_name)
+
+    remote_config_path = 'configs/{}.json'.format(dataset_name)
+    remote_dataset_path = 'datasets/{}.csv'.format(dataset_name)
+    config_blob = bucket.blob(remote_config_path)
+    dataset_blob = bucket.blob(remote_dataset_path)
+
+    if not config_blob.exists():
+        logger.error("File {} does not exist in the Google storage bucket.".format(remote_config_path))
+    if not dataset_blob.exists():
+        logger.error("File {} does not exist in the Google storage bucket.".format(remote_dataset_path))
+    if config_blob.exists() and dataset_blob.exists():
+        local_config_path = str(configs_dir / "{}.json".format(dataset_name))
+        logger.info("Downloading {} from the Google storage bucket to {}.".format(remote_config_path, local_config_path))
+        config_blob.download_to_filename(local_config_path)
```

Comment:
Should we enclose in these downloads in a try (they can throw `google.cloud.exceptions.NotFound`) or do you want the exception to propagate? (If we're downloading lots of things I think it might be better to keep going if one fails.)

Reply:
I think the `if` statement in Line 26 ensures that both objects are present in the bucket.

Reply:
Ok cool. I wasn't sure if this test was sufficient